packages/Webkul/API/Http/Controllers/Shop/CartController.php

```diff
@@ -86,47 +89,58 @@ public function get()
     /**
      * Store a newly created resource in storage.
      *
-     * @param  int  $id
-     * @return \Illuminate\Http\Response
+     * @param int $id
+     *
+     * @return \Illuminate\Http\JsonResponse
      */
-    public function store($id)
+    public function store($id): ?JsonResponse
     {
         if (request()->get('is_buy_now')) {
             Event::dispatch('shop.item.buy-now', $id);
         }
 
         Event::dispatch('checkout.cart.item.add.before', $id);
 
-        $result = Cart::addProduct($id, request()->except('_token'));
+        try {
+            $result = Cart::addProduct($id, request()->except('_token'));
 
-        if (! $result) {
-            $message = session()->get('warning') ?? session()->get('error');
+            if (is_array($result) && isset($result['warning'])) {
+                return response()->json([
+                    'error' => $result['warning'],
+                ], 400);
+            }
 
-            return response()->json([
-                'error' => session()->get('warning'),
-            ], 400);
-        }
+            if ($customer = auth($this->guard)->user()) {
+                $this->wishlistRepository->deleteWhere(['product_id' => $id, 'customer_id' => $customer->id]);
+            }
 
-        if ($customer = auth($this->guard)->user()) {
-            $this->wishlistRepository->deleteWhere(['product_id' => $id, 'customer_id' => $customer->id]);
-        }
+            Event::dispatch('checkout.cart.item.add.after', $result);
 
-        Event::dispatch('checkout.cart.item.add.after', $result);
+            Cart::collectTotals();
 
-        Cart::collectTotals();
+            $cart = Cart::getCart();
 
-        $cart = Cart::getCart();
+            return response()->json([
+                'message' => __('shop::app.checkout.cart.item.success'),
+                'data'    => $cart ? new CartResource($cart) : null,
+            ]);
+        } catch (Exception $e) {
+            Log::error('API CartController: ' . $e->getMessage(),
+                ['productID' => $id, 'cartID' => cart()->getCart() ?? 0]);
```

Comment:
since the context-information are shown as json inside the log It should be `product_id` and `cart_id`

Reply:
True, too. And done :) 

Thank you for reviewing!